src/Cake/Scripting/DebugScriptHost.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Cake.Core;
+using Cake.Core.Diagnostics;
+using Cake.Core.Scripting;
+
+namespace Cake.Scripting
+{
+    /// <summary>
+    /// The script host used to debug Cake scripts.
+    /// </summary>
+    public sealed class DebugScriptHost : ScriptHost
+    {
+        private readonly ICakeReportPrinter _reportPrinter;
+        private readonly ICakeLog _log;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DebugScriptHost"/> class.
+        /// </summary>
+        /// <param name="engine">The engine.</param>
+        /// <param name="context">The context.</param>
+        /// <param name="reportPrinter">The report printer.</param>
+        /// <param name="log">The log.</param>
+        public DebugScriptHost(
+            ICakeEngine engine,
+            ICakeContext context,
+            ICakeReportPrinter reportPrinter,
+            ICakeLog log) : base(engine, context)
+        {
+            _reportPrinter = reportPrinter;
+            _log = log;
+        }
+
+        /// <summary>
+        /// Runs the specified target.
+        /// </summary>
+        /// <param name="target">The target to run.</param>
+        /// <returns>The resulting report.</returns>
+        public override CakeReport RunTarget(string target)
+        {
+            var strategy = new DefaultExecutionStrategy(_log);
+            var message = "Attach debugger to process {0} to continue";
+            var pid = System.Diagnostics.Process.GetCurrentProcess().Id;
+
+            _log.Information("Performing debug...");
+            _log.Information(Verbosity.Quiet, string.Format(message, pid));
+
+            while (!System.Diagnostics.Debugger.IsAttached)
```

Comment:
@mholo65 Sorry for not answering this until now but I was busy having a wonderful weekend in your lovely country visiting Helsinki and was completely disconnected :smile:

I think it would make sense here to add a `IDebugger` abstraction.

```csharp
public interface IDebugger
{
  int GetProcessId();
  bool WaitForAttach(TimeSpan timeout);
}
```